airbyte/_executors/python.py

```diff
@@ -193,6 +208,71 @@ def get_installed_version(
 
             return None
 
+    @lru_cache(maxsize=128)
+    def _get_pypi_python_requirements(self, package_name: str) -> str | None:
+        """Get the requires_python field from PyPI for a package.
+
+        Args:
+            package_name: The PyPI package name to check
+
+        Returns:
+            The requires_python string from PyPI, or None if unavailable
+
+        Example:
+            For airbyte-source-hubspot, returns "<3.12,>=3.10"
+        """
+        if AIRBYTE_OFFLINE_MODE:
+            return None
+
+        try:
+            url = f"https://pypi.org/pypi/{package_name}/json"
+            response = requests.get(
+                url=url,
+                headers={"User-Agent": f"PyAirbyte/{get_version()}"},
+                timeout=10,
+            )
+            response.raise_for_status()
+
+            data = response.json()
+            return data.get("info", {}).get("requires_python")
+
+        except Exception:
+            return None
```

Comment:
This needs a clear comment to explain why we are being so dismissive of exception handling.